data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetEth1Data.java

```diff
@@ -0,0 +1,101 @@
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.function.Function;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.teku.api.ChainDataProvider;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.response.v1.teku.GetEth1DataResponse;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.AsyncApiResponse;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.datastructures.blocks.Eth1Data;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
+import tech.pegasys.teku.validator.coordinator.Eth1DataCache;
+
+public class GetEth1Data extends MigratingEndpointAdapter {
+
+  public static final String ROUTE = "/teku/v1/beacon/state/eth1data";
+
+  private static final SerializableTypeDefinition<Eth1Data> ETH1DATA_RESPONSE_TYPE =
+      SerializableTypeDefinition.<Eth1Data>object()
+          .name("GetEth1DataResponse")
+          .withField("data", Eth1Data.SSZ_SCHEMA.getJsonTypeDefinition(), Function.identity())
+          .build();
+
+  private final ChainDataProvider chainDataProvider;
+  private final Eth1DataCache eth1DataCache;
+
+  public GetEth1Data(final DataProvider dataProvider, final Eth1DataCache eth1DataCache) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getEth1Data")
+            .summary("Get new Eth1Data")
+            .description("Eth1Data we'd include in a new block if we are a proposer.")
+            .tags(TAG_TEKU)
+            .response(SC_OK, "Request successful", ETH1DATA_RESPONSE_TYPE)
+            .build());
+    this.chainDataProvider = dataProvider.getChainDataProvider();
+    this.eth1DataCache = eth1DataCache;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get new Eth1Data",
+      description = "Eth1Data we'd include in a new block if we are a proposer.",
+      tags = {TAG_TEKU},
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = GetEth1DataResponse.class)),
+        @OpenApiResponse(status = RES_NOT_FOUND),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR)
+      })
+  @Override
+  public void handle(@NotNull final Context ctx) throws Exception {
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(final RestApiRequest request) throws JsonProcessingException {
+    request.respondAsync(
+        chainDataProvider
+            .getBeaconStateAtHead()
+            .thenApply(
+                maybeStateAndMetadata -> {
+                  if (maybeStateAndMetadata.isEmpty()) {
+                    return AsyncApiResponse.respondNotFound();
```

Comment:
The `EndpointMetadata` doesn't declare not found as a valid response type - need to add it in and should add a test to confirm the 404 response is sent correctly.